Add investigate and timeline page (#25889)
DataDog/documentation: content/en/service_management/incident_management/investigate/timeline.md

---
title: Timeline
further_reading:
- link: "/service_management/incident_management/investigate/"
  tag: "Documentation"
  text: "Investigate incidents"
---

## Overview

{{< img src="/service_management/incidents/investigate/timeline/timeline_tab.png" alt="Example incident showing the Timeline tab" style="width:100%;" >}}

The Incident Timeline is the primary source of information for the work done during an incident. As actions are performed, new cells are added to the timeline in chronological order to capture the changes made, the person who made the change, and the time the changes were made.

> By default, timeline cells are sorted in `oldest first` order, but this can be changed to `newest first` using the button at the top of the timeline.

Reviewer: ```suggestion
By default, timeline cells are sorted in `oldest first` order, but you can change it to `newest first` using the button at the top of the timeline.
```

## Content types

Each cell has its own content type that indicates the kind of information the cell contains:

|  Content type      | Description                                                                                            |
| ------------------ | ------------------------------------------------------------------------------------------------------ |

> | Responder note     | A note manually written by an incident responder. Responder notes have the following sub-types:<br>- *Graph*: The responder note contains one or more Datadog graphs<br>- *Link*: The responder note contains a hyperlink<br>- *Code*: The responder note contains text wrapped in Markdown syntax for code blocks

Reviewer: Does a responder note always have to be of one of the subtypes, or is there just a "plain" responder note type?

Author: You can have a plain note, within a note cell you can add graphs and links. If you don't add anything, it'd just be a responder note.

| Incident update    | Any changes made to an incident's properties (including status and severity) or its impact.
| Integration update | Any changes made through the Incident Management product's [integrations][1].
| Task               | Any changes made to incident tasks in the Remediation section of the Incident Details page.
| Notification sent  | An update when a manual notification is sent by an incident responder.

### Responder notes

> Add responder notes directly to the timeline using the text box just underneath the tabs for switching between the different sections of the Incident Details page. Customize the timestamp of the responder note at creation time to capture important information that was relevant at an earlier point in time in the chronological order of the timeline. 
> 
> For responder notes you author, you can edit the content or timestamp, or delete the note entirely. You can also copy a link to a specific cell to share with teammates. You can also add responder notes [to the timeline from Slack][2].

Reviewer: ```suggestion
Add responder notes directly to the timeline using the text box underneath the section tabs of the Incident Details page. You can also add responder notes [to the timeline from Slack][2]. You can customize the timestamp of the responder note at creation time to capture important information that was relevant at an earlier point in time in the chronological order of the timeline. 

For responder notes you author, you can edit the content or timestamp, or delete the note entirely. You can also copy a link to a specific cell to share with teammates. 
```

### Graph cells

> Graph definitions are stored using share URLs for graphs if enabled in your [Organization Settings][3]. When a graph cell is added to the timeline, it has full interactive hover states as found in Dashboards, Notebooks, and other pages. After 24 hours of being added to the timeline, the graphs are replaced with static images capturing what the graph was displaying. This is to ensure that graphs that are displaying short retention data have backup images captured even after the live data in the graphs has expired.

Reviewer: ```suggestion
Graph definitions are stored using share URLs for graphs if enabled in your [Organization Settings][3]. For 24 hours after a graph cell is added to the timeline, it has the same full interactive hover states found in Dashboards, Notebooks, and other pages. After 24 hours in the timeline, the graph is replaced with static images capturing what the graph was displaying. This is to ensure that graphs with data that has short retention have backups you can view after the live data for the graphs expires.
```

### Images

To upload an image to be hosted by Datadog, drop an image file into the text box field above the timeline. This adds the image as an individual cell in the timeline.

You can also add an image to an existing cell:  
{{< img src="/service_management/incidents/investigate/timeline/timeline_cell_add_image.png" alt="Your image description" style="width:100%;" >}}
1. Click the pencil icon to edit a cell.  
2. Click the image icon and locate the image in your file directory.  
3. You can use any of the following options to upload an image to be hosted by Datadog:  

>     * Drop an image file into the upload area  
>     * Click **Choose File** and locate the image in your file directory  
>     * Paste a publicly accessible URL for the image

Reviewer: ```suggestion
    * Drop an image file into the upload area.
    * Click **Choose File** and locate the image in your file directory.
    * Paste a publicly accessible URL for the image.
```
